File: eShop_CleanArch/Domain/Entities/Product.cs
using Domain.Abstract;
using Domain.Entities.ValueObjects;

namespace Domain.Entities;

public class Product : Entity<Guid>
{
    public string Name { get; set; }
    public string Brand { get; set; }
    public string Img { get; set; }
    public Money Price { get; set; } = new(0, "TRY");
    public int Quantity { get; set; }
    public ProductDetail? ProductDetail { get; set; }
    public ProductDiscount? ProductDiscount { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsFeatured { get; set; } = false;
    public bool IsDeleted { get; set; } = false;
    public ICollection<ProductCategory> ProductCategories { get; set; } = new List<ProductCategory>();
    //public ICollection<Review> Reviews { get; set; } = new List<Review>();

}

File: eShopClient/src/app/features/models/product.ts
import { Price } from "./money";
import { ProductDetailModel } from "./productDetail";
import { ProductDiscountModel } from "./productDiscount";


 export interface ProductModel {
	id:string;
	name: string;
	brand: string;
	img: string;
	quantity: number;
	productDetail: ProductDetailModel;
	price: Price;
	isActive: boolean 
    isDeleted: boolean 
	createAt: Date;
	productCategories: ProductCategory[];
	cartId:string;
    wishListId: string;
	isFavorite?: boolean;
	productDiscount: ProductDiscountModel;
	 

}



export interface ProductCategory {
	categoryId: string;
	categoryName: string;
} 


